File: backend/src/agent_86/domain/models/message.py
from dataclasses import dataclass, field
from datetime import datetime
from typing import Any, Literal


MessageRole = Literal["system", "user", "assistant", "tool"]


@dataclass
class Message:
    id: str
    session_id: str
    user_id: str
    role: MessageRole
    content: str
    metadata: dict[str, Any] = field(default_factory=dict)
    created_at: datetime | None = None

File: frontend/src/types/message.ts
export type MessageRole = "system" | "user" | "assistant" | "tool";

export interface Message {
  id: string;
  session_id: string;
  user_id: string;
  role: MessageRole;
  content: string;
  metadata: Record<string, unknown>;
  created_at: string;
}

export interface CreateMessageRequest {
  role: MessageRole;
  content: string;
  metadata?: Record<string, unknown>;
}